app/renderer/components/Session/Session.js

```diff
@@ -44,124 +24,113 @@ export default class Session extends Component {
   }
 
   componentWillMount () {
-    const {setLocalServerParams, getSavedSessions, setSavedServerParams, getRunningSessions} = this.props;
+    const {setLocalServerParams, getSavedSessions, setSavedServerParams, setVisibleProviders, getRunningSessions} = this.props;
     (async () => {
-      this.checkProvidersVisibility();
       await getSavedSessions();
       await setSavedServerParams();
       await setLocalServerParams();
+      await setVisibleProviders();
       getRunningSessions();
     })();
   }
 
-  async checkProvidersVisibility () {
-    const visibleProviders = {};
-    for (let [providerName, provider] of _.toPairs(CloudProviders)) {
-      if (await provider.isVisible()) {
-        visibleProviders[providerName] = true;
-      }
+  handleSelectServerTab (tab) {
+    const {changeServerType, addCloudProvider} = this.props;
+    if (tab === 'addCloudProvider') {
+      addCloudProvider();
+      return;
     }
-    this.setState({
-      ...this.state,
-      visibleProviders
-    });
+    changeServerType(tab);
```

Comment:
why do we call the entity ServerType when changing and CloudProvided when adding?

Reply:
`serverType` encompasses Cloud Providers and local